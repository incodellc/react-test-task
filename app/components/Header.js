import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

export const Header = ({ tickers }) => (
    <header className="header">
        <nav>
            <ul>
                {tickers.map((ticker) => (
                    <li key={ticker}>
                        <NavLink to={`/${ticker}`} activeClassName="active">
                            {ticker}
                        </NavLink>
                    </li>
                ))}
            </ul>
        </nav>
    </header>
);

Header.propTypes = {
    tickers: PropTypes.arrayOf(PropTypes.string).isRequired,
};
