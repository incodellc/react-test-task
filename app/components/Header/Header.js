import React from 'react';

import  SetIntervalForm from '../SetIntevalForm/SetIntevalForm';

const Header = () => {
    return (
        <div className="row">
            <h1 className="col-lg-4 text-center">Stock Blotter</h1>
            <SetIntervalForm/>
        </div>
    );
};

export default Header;

