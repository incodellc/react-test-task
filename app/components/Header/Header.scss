@import "../variables";

.stock-wrapper {
  &__header {
      width: 100%;
      height: 100px;
      display: flex;
      align-items: center;

    @media (min-width: $layout-breakpoint-small) {
      height: 200px;
    }

    &-logo {
      padding-left: 10%;
    }

    &-img {
      width: 100%;
      max-width: 300px;
      height: auto;

      @media (min-width: $layout-breakpoint-small) {
        max-width: 400px;
      }
    }
  }
}

@media (min-width: $layout-breakpoint-small) {

}
