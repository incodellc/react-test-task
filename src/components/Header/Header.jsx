import React from "react";
import { Container } from "react-bootstrap";

import AppDropdown from "../AppDropdown";
import ConnectButton from "../ConnectButton";

const Header = props => {
    return (
        <div id="header-component" data-testid="header">
            <Container>
                <h1>Stock Blotter</h1>
                <div className="d-flex mt-2 mb-2 align-items-center">
                    <AppDropdown data-testid="app-dropdown"/>
                    <ConnectButton />
                </div>
            </Container>
        </div>
    );
};

export default Header;
