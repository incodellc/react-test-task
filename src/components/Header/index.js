import Header from "./Header";
import { connect } from "react-redux";

export default connect(state => {
    return {
        nasdaqList: state.nasdaqList,
        nasdaqCurrent: state.nasdaqCurrent,
        connect: state.connect
    };
})(Header);;