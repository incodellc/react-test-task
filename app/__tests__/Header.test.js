import React from 'react';
import { shallow } from 'enzyme';
import Header from '../components/Header/Header';

describe('Header component test ', () => {
    const header = shallow(<Header />);

    it('should renders properly', () => {
        expect(header).toMatchSnapshot();
    });

    it ('should render text Stock Blotter', () => {
        expect(header.find('h1').text()).toEqual('Stock Blotter');
    });
});
