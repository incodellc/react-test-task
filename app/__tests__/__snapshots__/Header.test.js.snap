// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component test  should renders properly 1`] = `
<div
  className="row"
>
  <h1
    className="col-lg-4 text-center"
  >
    Stock Blotter
  </h1>
  <Connect(SetIntervalForm) />
</div>
`;
